Fix for Code quality improvements

huyuwei1996 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

huyuwei1996 committed:
Fix test for switchbot translate

(cherry picked from commit 8c3e0b7342b6185512b757d9bf2e87ea2402c490)

joostlek commented on homeassistant/components/switchbot/strings.json:
```diff
@@ -5,6 +5,9 @@
           "address": "Device address"
+        },
+        "data_description": {
+          "address": "The address of the device"
```
I think we can be clearer, what address?

joostlek commented on homeassistant/components/switchbot/strings.json:
```diff
@@ -14,20 +17,31 @@
           "password": "[%key:common::config_flow::data::password%]"
+        },
+        "data_description": {
+          "password": "The password for the device"
```
How did the user set up this password? Is it static?

joostlek commented on homeassistant/components/switchbot/strings.json:
```diff
@@ -14,20 +17,31 @@
           "password": "[%key:common::config_flow::data::password%]"
+        },
+        "data_description": {
+          "username": "SwitchBot account username/email",
```
Which one of the two?

joostlek commented on homeassistant/components/switchbot/strings.json:
```diff
@@ -14,20 +17,31 @@
+        },
+        "data_description": {
+          "username": "SwitchBot account username/email",
+          "password": "SwitchBot account password"
```
```suggestion
          "password": "The password of your SwitchBot account"
```

epenet committed:
Use new SsdpServiceInfo location in remaining components (#136053)

joostlek committed:
Bump yt-dlp to 2025.01.15 (#136072)

NoRi2909 committed:
Fix casing of "client" and "ID" in transmission integration (#136071)

edenhaus committed:
Bump deebot-client to 11.0.0 (#136073)

emontnemery committed:
Make the time for automated backups configurable (#135825)

* Make the time for automated backups configurable

* Store time as a string, use None to indicate default time

* Don't add jitter if the time is set by user

* Include time of next automatic backup in response to backup/info

* Update tests

* Rename recurrence to state

* Include scheduled backup time in backup/config/info response

* Address review comments

* Update cloud test

* Add test for store migration

* Address review comments

emontnemery committed:
Allow backup writer to update progress during restore (#135975)

* Allow backup writer to update progress during restore

* Clarify comment

epenet committed:
Use new ServiceInfo location in component tests (part 1) (#136057)

epenet committed:
Use new ServiceInfo location in component tests (part 2) (#136062)

epenet committed:
Use new ServiceInfo location in component tests (part 3) (#136064)

epenet committed:
Use new ServiceInfo location in component tests (part 4) (#136065)

NoRi2909 committed:
Improve user interface strings in opentherm_gw (#136078)

epenet committed:
Use new ServiceInfo location in core tests (#136067)

huyuwei1996 committed:
Update homeassistant/components/switchbot/strings.json

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

huyuwei1996 committed:
Update SwitchBot username description in strings.json

NoRi2909 committed:
Fix spelling of "ID" in hyperion user strings (#136082)

NoRi2909 committed:
Replace `targets` key with UI name 'Targets' in media_player.join action (#136063)

PaulSD committed:
Prevent pylint out-of-memory failures (#136020)

joostlek commented on homeassistant/components/switchbot/strings.json:
```diff
@@ -14,20 +17,31 @@
           "password": "[%key:common::config_flow::data::password%]"
+        },
+        "data_description": {
+          "username": "SwitchBot account username",
```
Let's have this also in a more descriptive way

abmantis committed:
Use fixtures for Network component tests (#135220)

emontnemery committed:
Always include SSL folder in backups (#136080)

cdce8p committed:
Improve conversation typing (#136084)

joostlek committed:
Set configuration url to overseerr instance (#136085)

edenhaus committed:
Bump uv to 0.5.21 (#136086)

cdce8p committed:
Update pylint to 3.3.3 and astroid to 3.3.8 (#136090)

joostlek committed:
Fix Overseerr event types translations (#136096)

joostlek committed:
Add diagnostics to Overseerr (#136094)

autinerd committed:
Enable Ruff B024 (#136088)

cereal2nd committed:
Bump velbusaio to 2025.1.1 (#136089)

sdb9696 committed:
Enable dynamic child devices for tplink module entities (#135822)

Add dynamic child device handling to tplink integration for module based entities. For child devices that could be added/removed to hubs.

This address the module based platforms. #135229 addressed feature based platforms.

thecode committed:
Bump aiowebostv to 0.5.0 (#136097)

andrewsayre committed:
Add integration_type to HEOS (#136105)

jbouwh committed:
Fix index in incomfort diagnostics generator (#136108)

autinerd committed:
Fix upload service response for google_photos (#136106)

joostlek committed:
Add icon to overseerr (#136110)

thecode committed:
Bump ayla-iot-unofficial to 1.4.5 (#136099)

joostlek committed:
Bump python-overseerr to 0.6.0 (#136104)

andrewsayre committed:
Raise exceptions in HEOS service actions (#136049)

* Raise errors instead of log

* Correct docstring typo

autinerd committed:
Enable Ruff B035 (#135883)

andrewsayre committed:
Remove yaml config fixture from HEOS tests (#136123)

bieniu committed:
Add additional entities for Shelly BLU TRV (#135244)

* Add valve position sensor

* Add valve position and external sensor temperature numbers

* Fix method name

* Better name

* Add remove condition

* Add calibration binary sensor

* Add battery and signal strength sensors

* Remove condition from ShellyRpcEntity

* Typo

* Add get_entity_class helper

* Add tests

* Use snapshots in tests

catsmanac committed:
Add scheduled envoy firmware checks to enphase_envoy coordinator (#136102)

* Add scheduled envoy firmware checks to enphase_envoy coordinator

* Set firmware scantime to 4 hours and split test in 2

gjohansson-ST committed:
Bump holidays to 0.65 (#136122)

jpelgrom committed:
Bump letpot to 0.3.0 (#136133)

allenporter committed:
Add myself to Roborock codeowners (#136134)

catsmanac committed:
Add data descriptions for enphase_envoy config flows. (#136120)

cdnninja committed:
Remove not needed warning in Z-Wave (#136006)

* Remove unneeded logging

* ruff correction

huyuwei1996 committed:
Update SwitchBot strings to clarify address and password descriptions
